services/Donation.js

```diff
@@ -60,6 +60,87 @@ class DonationService {
       return false;
     }
   }
+
+  /**
+   * @description get total donations
+   */
+  static async getTotal(startDate, endDate) {
+    try {
+      const matchQuery = {};
+      if (startDate & endDate) {
+        matchQuery.createdAt = { $gte: startDate, $lte: endDate };
+      }
+
+      const totalDonations = await Donation.aggregate([
+        { $match: matchQuery },
+        { $group: { _id: null, totalAmount: { $sum: "$transaction.amount" } } }
+      ]);
+
+      if (totalDonations.length === 0) {
+        return { totalAmount: 0 };
+      }
+
+      return { totalAmount: totalDonations[0].totalAmount };
+    } catch (e) {
+      throw new Error("Internal Server Error");
```

Comment:
Let's log an descriptive message with the exact error details so we can properly debug when we have an issue - then throw the error so it's caught by the calling service:

```
Logger.error('Error getting total donations', JSON.stringify(e))
throw e;
```